VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -25,17 +25,900 @@ if ($PSEdition -eq 'Desktop') {
+        if ($PSBoundParameters.ContainsKey("enable")) {
+
+            Write-Host "Enabling lockdown mode for each ESXi host in $cluster cluster"
+            foreach ($esxiHost in $esxiHosts) {
+                $currentLockdownMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                if ($currentLockdownMode -eq "lockdownDisabled") {
+                    ($esxiHost | Get-View).EnterLockdownMode()
+                    Write-Host "Changing lockdown mode for ESXi host $esxiHost from $currentLockdownMode to lockdownNormal."
+                    $newLockdownMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                    if ($lockdownMode -eq $newLockdownMode) {
+                        Write-Error "Unable to change lockdown mode for ESXi host $esxiHost from $currentLockdownMode to lockdownNormal. Lockdown mode is set to $newLockdownMode." -ErrorAction Stop}
+                } else {
+                    Write-Warning "Lockdown mode for ESXi host $esxiHost is already set to lockdownNormal: SKIPPED"
+                }
+            }
+        } 
+        
+        if ($PSBoundParameters.ContainsKey("disable")) {
+            Write-Host "Disabling lockdown mode for each ESXi host in $cluster cluster."
+            foreach ($esxiHost in $esxiHosts) {
+                $currentLockdownMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                if ($currentLockdownMode -ne "lockdownDisabled") {
+                    ($esxiHost | Get-View).ExitLockdownMode()
+                    Write-Host "Changing lockdown mode for ESXi host $esxiHost from $currentLockdownMode to lockdownDisabled."
+                    $newLockdownMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                    if ($currentLockdownMode -eq $newLockdownMode) {
+                        Write-Error "Unable to change lockdown mode for ESXi host $esxiHost from $currentLockdownMode to lockdownDisabled. Lockdown mode is set to $newLockdownMode." -ErrorAction Stop
+                    }
+                } else {
+                    Write-Warning "Lockdown mode for ESXi host $esxiHost is already set to lockdownDisabled: SKIPPED"
+                }
+            }
+        } 
+    }
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+}
+
+Export-ModuleMember -Function Set-ESXiLockdownMode
+
+
+Function Restart-ESXiHost {
+
+    <#
+        .SYNOPSIS
+        Restart the provided ESXi host and poll for it to come back online
+
+        .DESCRIPTION
+        This cmdlet triggers a restart the provided ESXi host and polls for it to come back online. 
+        Timeout value is in seconds.
+
+        .EXAMPLE
+        Restart-EsxiHost -esxiFqdn sfo01-m01-esx03.sfo.rainpole.io -user root -pass VMw@re1! -poll $true -timeout 1800 -pollInterval 30
+        This example restarts the provided esxi hosts and polls every 30 seconds till it comes back online. It will timeout after 1800 seconds. 
```

Comment:
```suggestion
        This example restarts an ESXi host and polls the connection availability every 30 seconds. It will timeout after 1800 seconds. 
```